lib/notifications-status-view.js

```diff
@@ -0,0 +1,76 @@
+const { CompositeDisposable, Disposable } = require('atom');
+const moment = require('moment');
+
+module.exports = class NotificationsStatusView {
+
+    constructor(statusBar, duplicateTimeDelay) {
+      this.count = 0;
+      this.duplicateTimeDelay = duplicateTimeDelay;
+      this.subscriptions = new CompositeDisposable;
+      this.render();
+      this.tile = statusBar.addRightTile({
+        item: this.element,
+        priority: 100
+      });
+      this.subscriptions.add(new Disposable(() => { this.tile.destroy(); }));
+      // TODO: uncomment next line when atom/atom#16074 is released
+      // this.subscriptions.add(atom.notifications.onDidClearNotifications(() => this.clear()));
+    }
+
+    render() {
+      this.number = document.createElement('div');
+      this.number.textContent = this.count;
+      this.number.addEventListener('animationend', (e) => {
+        if (e.animationName === 'new-notification') {
+          this.number.classList.remove('new-notification');
+        }
+      });
+
+      this.element = document.createElement('a');
+      this.element.classList.add('notifications-count', 'inline-block');
+      this.tooltip = atom.tooltips.add(this.element, {title: 'Notifications'});
+      let span = document.createElement('span');
+      span.appendChild(this.number);
+      this.element.appendChild(span);
+      this.element.addEventListener('click', () => {
+        atom.commands.dispatch(this.element, 'notifications:toggle-log')
+      });
+
+      let lastNotification = null;
```

Comment:
Isn't this already handled in https://github.com/UziTech/notifications/blob/5191c3264ea7e2e43c4b2cb3aec45d650111a8c9/lib/main.coffee#L114-L120?

I noticed I missed this similar piece of code in my review of the notifications log, sorry for not mentioning it earlier.

Reply:
yes this handles adding notifications that have happened before initializing the statusbar.

Without this code here, if the user disables and re-enables the statusbar package the count would be set to 0 even if there are notifications in the log.

Reply:
I guess I could just pass the notificationsLog item count and the last notification to get the type and message for the tooltip